pkg/util/mount/mount_linux.go

```diff
@@ -892,7 +892,14 @@ func doCleanSubPaths(mounter Interface, podDir string, volumeName string) error
 		fullContainerDirPath := filepath.Join(subPathDir, containerDir.Name())
 		subPaths, err := ioutil.ReadDir(fullContainerDirPath)
 		if err != nil {
-			return fmt.Errorf("error reading %s: %s", fullContainerDirPath, err)
+			glog.Errorf("error reading %s: %s", fullContainerDirPath, err)
+			if isCorruptedMnt(err) {
+				for i := 0; pathExists(path.Join(fullContainerDirPath, fmt.Sprint(i))); i++ {
```

Comment:
could you add comment here? why would you loop from 0 to n here? what's the expected count usually?

Reply:
Thanks for your review.
Comment added!